ipv-core/src/main/java/com/ibm/whc/deid/models/City.java

```diff
@@ -23,24 +25,65 @@
   /**
    * Instantiates a new City.
    *
-   * @param name the name
+   * @param name the city name
+   * @param latitude the latitude in string form
+   * @param longitude the longitude in string form
+   * @param countryCode the country code
+   * @param nameCountryCode the name country code
+   * 
+   * @throws IllegalArgumentException if any of the input values is null, whitespace, or invalid.
+   */
+  public City(String name, String latitude, String longitude, String countryCode,
+      String nameCountryCode) {
+    this.name = name;
+    this.countryCode = countryCode;
+    this.nameCountryCode = nameCountryCode;
+    init();
+
+    this.latitudeLongitude = new LatitudeLongitude(latitude, longitude);
+
+    this.key = name.toUpperCase();
+  }
+
+  /**
+   * Instantiates a new City.
+   *
+   * @param name the city name
    * @param latitude the latitude
    * @param longitude the longitude
    * @param countryCode the country code
    * @param nameCountryCode the name country code
+   * 
+   * @throws IllegalArgumentException if any of the input values is null, whitespace, or invalid.
    */
-  public City(String name, Double latitude, Double longitude, String countryCode,
+  public City(String name, double latitude, double longitude, String countryCode,
       String nameCountryCode) {
     this.name = name;
     this.countryCode = countryCode;
     this.nameCountryCode = nameCountryCode;
+    init();
 
     this.latitudeLongitude =
         new LatitudeLongitude(latitude, longitude, LatitudeLongitudeFormat.DECIMAL);
 
     this.key = name.toUpperCase();
   }
 
+  private void init() {
+    if (name == null || name.trim().isEmpty()) {
+      throw new IllegalArgumentException(
+          Messages.getMessage(LogCodes.WPH1010E, String.valueOf(name), "city name"));
+    }
+    if (countryCode == null || countryCode.trim().isEmpty()) {

```

Comment:
Docs says that some columns are optional :
city
File name: city_list.csv Contains a list of cities and related information.

Mandatory column: name

Optional columns: latitude, longitude, country code, population

Reply:
I will investigate and re-instate the optional nature of the documented columns as part of the next delivery for this story.